### PR TITLE
Access logging

### PR DESCRIPTION
**Pull request by EItanya**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/814

### CODE REVIEW

File: projects/gloo/api/v1/plugins.proto [hunk 1]
```diff
@@ -37,14 +38,15 @@ import "google/protobuf/duration.proto";
 // Note to developers: new Listener Plugins must be added to this struct
 // to be usable by Gloo.
 message ListenerPlugins {
-    // Future sight for matching
```
Comment (yuval-k) [inline projects/gloo/api/v1/plugins.proto]: no need to remove this comment

File: projects/gloo/api/v1/plugins/als/als.proto [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+message AccessLog {
+    // filters for this listeners access logger.
+    Filter filter = 1;
+
+    // type of Access Logging service to implement
+    oneof OutputDestination {
+        FileSink file_sink = 2;
+        GrpcService grpc_service = 3;
```
Comment (yuval-k) [inline projects/gloo/api/v1/plugins/als/als.proto]: you can remove this until it is implemented, and still keep the oneof.

File: projects/gloo/api/v1/plugins/als/als.proto [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+// Contains various settings for Envoy's access logging service.
+// See here for more information: https://www.envoyproxy.io/docs/envoy/latest/api-v2/config/filter/accesslog/v2/accesslog.proto#envoy-api-msg-config-filter-accesslog-v2-accesslog
+message AccessLoggingService {
+    repeated AccessLog access_log = 1;
+}
+
+message AccessLog {
+    // filters for this listeners access logger.
+    Filter filter = 1;
```
Comment (yuval-k) [inline projects/gloo/api/v1/plugins/als/als.proto]: Filter is empty; do we plan to implement it in this PR? if no, remove or comment

File: projects/gloo/pkg/translator/utils.go [hunk 4]
```diff
@@ -42,6 +42,17 @@ func NewFilterWithConfig(name string, config proto.Message) (envoylistener.Filte
 	return s, nil
 }
 
+func NewConfig(config proto.Message) (*types.Struct, error) {
```
Comment (yuval-k) [inline projects/gloo/pkg/translator/utils.go]: i'm pretty sure this utility function already exists

File: projects/gloo/pkg/xds/envoy_server.go [hunk 5]
```diff
@@ -98,3 +98,11 @@ func (s *envoyServer) DeltaClusters(_ v2.ClusterDiscoveryService_DeltaClustersSe
 func (s *envoyServer) DeltaRoutes(_ v2.RouteDiscoveryService_DeltaRoutesServer) error {
 	return errors.New("not implemented")
 }
+
+func (s *envoyServer) DeltaEndpoints(v2.EndpointDiscoveryService_DeltaEndpointsServer) error {
+	panic("implement me")
```
Comment (yuval-k) [inline projects/gloo/pkg/xds/envoy_server.go]: above function returns `errors.New("not implemented")`. let's do the same behavior on these as well.

File: test/e2e/gateway_test.go [hunk 6]
```diff
@@ -25,7 +29,7 @@ import (
 	"github.com/solo-io/solo-kit/pkg/api/v1/resources/core"
 )
 
-var _ = Describe("Gateway", func() {
+var _ = FDescribe("Gateway", func() {
```
Comment (yuval-k) [inline test/e2e/gateway_test.go]: unf

File: test/services/envoy.go [hunk 7]
```diff
@@ -310,17 +310,22 @@ func (ei *EnvoyInstance) Clean() error {
 	return nil
 }
 
+// func (ei *EnvoyInstance) GetLogs() io.Reader  {
```
Comment (yuval-k) [inline test/services/envoy.go]: delete

File: projects/gloo/pkg/plugins/als/plugin.go [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+
+func (p *Plugin) ProcessListener(params plugins.Params, in *v1.Listener, out *envoyapi.Listener) error {
+	if in.GetPlugins() == nil {
+		return nil
+	}
+	alSettings := in.GetPlugins()
+	if alSettings.AccessLoggingService == nil {
+		return nil
+	}
+	// Use a switch here to support TCP in the future
```
Comment (yuval-k) [inline projects/gloo/pkg/plugins/als/plugin.go]: tcp seems to be supported. comment can be deleted?

File: projects/gloo/pkg/translator/utils.go (hunk 4 above)
Comment (EItanya) [reply projects/gloo/pkg/translator/utils.go]: There's a function which does this for filters specifically, but I needed a more generic one

**Pull request merged**